test/std/algorithms/alg.sorting/alg.binary.search/binary.search/binary_search.pass.cpp

```diff
@@ -0,0 +1,103 @@
+// <algorithm>
+
+// template<ForwardIterator Iter, class T>
+//   constexpr bool      // constexpr after C++17
+//   binary_search(Iter first, Iter last, const T& value);
+
+#include "oneapi_std_test_config.h"
+#include "test_iterators.h"
+#include "checkData.h"
+
+#include <iostream>
+
+#ifdef USE_ONEAPI_STD
+#    include _ONEAPI_STD_TEST_HEADER(algorithm)
+#    include _ONEAPI_STD_TEST_HEADER(iterator)
+namespace s = oneapi_cpp_ns;
+#else
+#    include <algorithm>
+#    include <iterator>
+namespace s = std;
+#endif
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
+constexpr sycl::access::mode sycl_write = sycl::access::mode::write;
+
+template <class Iter, class T>
+bool
+test(Iter first, Iter last, const T& value, bool x)
+{
+    return (s::binary_search(first, last, value) == x);
+}
+
+class KernelBSearchTest1;
+class KernelBSearchTest2;
+class KernelBSearchTest3;
+class KernelBSearchTest4;
+
+template <typename Iter, typename KC>
+void
+kernel_test()
+{
+    sycl::queue deviceQueue = TestUtils::get_test_queue();
+    sycl::cl_bool ret = false;
+    sycl::range<1> numOfItems{1};
+    sycl::buffer<sycl::cl_bool, 1> buffer1(&ret, numOfItems);
+    const unsigned N = 1000;
+    const int M = 10;
+    int host_vbuf[N];
+    int x = 0;
+    for (size_t i = 0; i < N; ++i)
+    {
+        host_vbuf[i] = x;
```

Comment:
This loop looks like it's setting values from 0 to `(M-1)` by resetting the `x` variable once it gets to M. Could you instead change this line to 
```suggestion
        host_vbuf[i] = i % M;
```

Reply:
I would go even further, and changed this loop to generate sorted data, instead of applying std::sort later.